backend/src/src-cvesearch/src/main/java/org/eclipse/sw360/cvesearch/datasource/matcher/ModifiedLevenshteinDistance.java

```diff
@@ -9,13 +9,61 @@
  */
 package org.eclipse.sw360.cvesearch.datasource.matcher;
 
+import java.util.Optional;
+
 import static org.eclipse.sw360.datahandler.common.CommonUtils.nullToEmptyString;
 
 public class ModifiedLevenshteinDistance {
 
     public static Match levenshteinMatch(String needle, String haystack){
         return new Match(needle,
-                calculateModifiedLevenshteinDistance(needle,nullToEmptyString(haystack).toLowerCase().replace(' ', '_')));
+                calculateModifiedLevenshteinDistance(needle, nullToEmptyString(haystack).replace(' ', '_')));
+    }
+
+    public static class LevenshteinCost {
+
+        private final int d;
+        private final boolean consumed;
```

Comment:
Maybe, just maybe, this field and corresponding methods would better be called `matched`